src/pineko/evolve.py

```diff
@@ -309,32 +393,63 @@ def evolve_grid(
         )
         for mur2 in mur2_grid
     ]
-    # We need to use ekompatibility in order to pass a dictionary to pineappl
-    fktable = grid.evolve(
-        ekompatibility.pineappl_layout(operators),
-        xir * xir * mur2_grid,
-        alphas_values,
-        "evol",
-        order_mask=order_mask,
-        xi=(xir, xif),
-    )
+
+    def prepare(operator, items):
+        (q2, _), op = items
+        info = PyOperatorSliceInfo(
+            fac0=operator.mu20,
+            x0=operator.bases.inputgrid.raw,
+            pids0=basis_rotation.evol_basis_pids,
+            fac1=q2,
+            x1=operator.bases.targetgrid.raw,
+            pids1=operator.bases.targetpids,
+            pid_basis=PyPidBasis.Evol,
+        )
+        return (info, op.operator)
+
+    if operators2 is not None:
+        # check convolutions order
+        check_convolution_types(grid, operators1, operators2)
+        fktable = grid.evolve_with_slice_iter2(
+            map(lambda it: prepare(operators1, it), operators1.items()),
+            map(lambda it: prepare(operators2, it), operators2.items()),
```

Comment:
On second though,yes, I guess we can also merge the iteration into `prepare`. E.g. we can do
```py
def prepare(operators):
   # blub
   for it in operators.items():
      # bla
      yield x,y
```

Reply:
I feel we are going in loop here, it was basically what we had before...

Reply:
mmm, are we? I think before we were creating a list and return that list - and this was inefficient, because we iterated everywhere. The way I suggest now is only iterating once thanks to [yield](https://docs.python.org/3/reference/expressions.html#yieldexpr).

Reply:
solved in [495ad80](https://github.com/NNPDF/pineko/pull/181/commits/495ad80cd41030c9eaa4abee109e0f62376376a8)